sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/SqaleXmlLoader.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * SonarSource :: .NET :: Shared library
+ * Copyright (C) 2014-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.dotnet.shared.plugins;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.nio.charset.StandardCharsets;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamException;
+import org.codehaus.staxmate.SMInputFactory;
+import org.codehaus.staxmate.in.SMHierarchicCursor;
+import org.codehaus.staxmate.in.SMInputCursor;
+import org.sonar.api.server.debt.DebtRemediationFunction;
+import org.sonar.api.server.rule.RulesDefinition.DebtRemediationFunctions;
+import org.sonar.api.server.rule.RulesDefinition.NewRepository;
+import org.sonar.api.server.rule.RulesDefinition.NewRule;
+
+/**
+ * Copied from sslr-squid bridge to cut dependency and remove Guava dependency
+ */
+public class SqaleXmlLoader {
+
+  private NewRepository repository;
+
+  public SqaleXmlLoader(NewRepository repository) {
+    this.repository = repository;
+  }
+
+  public static void load(NewRepository repository, String xmlResourcePath) {
+    new SqaleXmlLoader(repository).loadXmlResource(xmlResourcePath);
+  }
+
+  public void loadXmlResource(String resourcePath) {
+    try (InputStream is = SqaleXmlLoader.class.getResourceAsStream(resourcePath); Reader reader = new BufferedReader(new InputStreamReader(is, StandardCharsets.UTF_8))) {
+      XMLInputFactory xmlFactory = XMLInputFactory.newInstance();
+      xmlFactory.setProperty(XMLInputFactory.IS_COALESCING, Boolean.TRUE);
+      xmlFactory.setProperty(XMLInputFactory.IS_NAMESPACE_AWARE, Boolean.FALSE);
+      // just so it won't try to load DTD in if there's DOCTYPE
+      xmlFactory.setProperty(XMLInputFactory.SUPPORT_DTD, Boolean.FALSE);
+      xmlFactory.setProperty(XMLInputFactory.IS_VALIDATING, Boolean.FALSE);
+      SMInputFactory inputFactory = new SMInputFactory(xmlFactory);
+      try {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1141)